git_merge_range/args.py

```diff
@@ -0,0 +1,157 @@
+# SPDX-License-Identifier: LGPL-2.1+
+from __future__ import annotations
+
+import argparse
+import subprocess
+import typing as ty
+
+from . import gitutil
+
+
+class Args(ty.NamedTuple):
+    current: gitutil.NormalizedRevRange
+    base: gitutil.NormalizedRevRange
+    other: gitutil.NormalizedRevRange
+    target: ty.Optional[str]
+    onto: ty.Optional[str]
+    todo_append: ty.List[str] = []
+
+
+parser = argparse.ArgumentParser(
+    description="""
+git-merge-range implements a 3-way-ish merge of commit ranges.
+
+The arguments current, base and other have similar semantics for those of
+git-merge-file: changes from the base to the other range are incorporated into
+the current range.
+
+This is implemented by a "diff-wise" matching of commits from the three ranges
+(currently using git-range-diff) and merging the commit sequences into a
+rebase-todo that is fed to an interactive rebase for editing.
+
+Conflicts are presented in the rebase-todo with the usual conflict markers.
+Conflicts usually arise when both current and other ranges change common
+subranges of base with non-matching commits.
+
+Each "current", "base" and "other" argument can be either a revision range or a
+single revision. In the case of the latter, the single revision is taken as the
+head and the merge base (git merge-base --octopus ...) of the three ranges'
+heads is used as default base.
```

Comment:
I'm more a example-driven person and like the ascii art examples in git-log(1) / git-merge(1). Could you add one for this command?

Reply:
Yep. I'll try to do it when I get some time.